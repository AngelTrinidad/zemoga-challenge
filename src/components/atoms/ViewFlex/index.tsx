import React, {FC} from 'react';
import {View, ViewProps} from 'react-native';

import styles from './styles';

import {VIEW_CENTER_STYLES} from '~constants/styles';

interface Props extends ViewProps {
  center?: boolean;
}

const ViewFlex: FC<Props> = ({style, children, center, ...props}) => {
  return (
    <View style={[styles.view, center && VIEW_CENTER_STYLES, style]} {...props}>
      {children}
    </View>
  );
};

export default ViewFlex;
