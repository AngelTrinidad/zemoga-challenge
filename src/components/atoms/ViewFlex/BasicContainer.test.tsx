import React from 'react';
import {Text} from 'react-native';

import ViewFlex from '.';

import {render} from '~helpers/testing';

describe('<ViewFlex />', () => {
  let component: any;

  beforeEach(() => {
    component = render(
      <ViewFlex>
        <Text>Hello</Text>
      </ViewFlex>,
    );
  });

  it('should render correctly', () => {
    expect(component).toBeDefined();
  });

  it('should has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
