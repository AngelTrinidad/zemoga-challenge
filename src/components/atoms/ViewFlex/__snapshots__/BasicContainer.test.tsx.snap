// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewFlex /> should has same snapshot 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
      undefined,
    ]
  }
>
  <Text>
    Hello
  </Text>
</View>
`;
